lib/roby/app.rb

```diff
@@ -1041,6 +1048,33 @@ def base_setup
             @base_setup_done = true
         end
 
+        def lock_log_dir
+            return if log_dir_locked?
+
+            temp_lock_file = File.join(log_dir, "#{LOCK_FILE_EXT}.tmp")
+            final_lock_file = File.join(log_dir, LOCK_FILE_EXT)
+
+            @lock_file = File.open(temp_lock_file, File::RDWR | File::CREAT, 0o644)
+            if @lock_file.flock(File::LOCK_EX | File::LOCK_NB)
+                File.rename(temp_lock_file, final_lock_file)
+                @lock_file.close
+                @lock_file = File.open(final_lock_file, File::RDWR | File::CREAT, 0o644)
+            end
```

Comment:
I would instead of attributing a tmp file to @ lock_file, I would create a tmp_file, as this is not yet the final file:

```suggestion
            return if log_dir_locked?

            temp_lock_path = File.join(log_dir, "#{LOCK_FILE_EXT}.tmp")
            final_lock_path = File.join(log_dir, LOCK_FILE_EXT)

            temp_lock_file = File.open(temp_lock_path, File::RDWR | File::CREAT, 0o644)
            if temp_lock_file.flock(File::LOCK_EX | File::LOCK_NB)
                File.rename(temp_lock_path, final_lock_path)
                temp_lock_file.close
                @lock_file = File.open(final_lock_path, File::RDWR | File::CREAT, 0o644)
            end
```

Reply:
I could be missing something

Reply:
I don't think this would make much of a difference. Did you suggest that to add more clarity to the code?